app/api/services/user_claims.py

```diff
@@ -0,0 +1,54 @@
+from datetime import datetime
+from app.api.helpers import Service
+from app import db
+from app.api.helpers import generate_random_token
+from app.models import UserClaims
+
+
+class UserClaimsService(Service):
+    __model__ = UserClaims
+
+    def __init__(self, *args, **kwargs):
+        super(UserClaimsService, self).__init__(*args, **kwargs)
+
+    # records a claim of ownership of a user email address
+    def make_claim(self, type=None, email_address=None, data=None):
+        if not data:
+            data = {}
+        saved = False
+        if email_address and data:
+            token = generate_random_token()
+            claim = UserClaims(type=type, token=token, email_address=email_address, data=data)
+            db.session.add(claim)
+            db.session.commit()
+            saved = True
+        return token if saved else None
+
+    # returns a claim instance without modifying anything
+    def get_claim(self, type=None, token=None, email_address=None, claimed=None):
```

Comment:
I am wondering if it would be easier to use find in all places that uses this instead of using creating this function.